### PR TITLE
Utilize datafile

### PR DESCRIPTION
**Pull request by Hoverbear**
Moves a considerable amount of logic into `_data/` to enable automation and enables JS-less tooltips on the frontpage.

### CODE REVIEW

**Comment by marumari** [comment]
@lgarron: you can see this on the "new" badssl server @ 104.197.54.253, running inside docker.

Make sure to grab the root: http://http.badssl.com/certs/badssl-root.pem

By and large, I really like the new interface.  Some thoughts:

- Headers should probably be renamed: Certs -> Certificates, Cipher-Suites -> Cipher Suites, Known-Bad -> Known Bad, Misc -> Miscellaneous, Mixed -> Mixed Content
- The popout behavior is a little bit annoying, because of how things expand and contract.  If you have one tooltip open and then open another, your mouse will stop hovering over a link when the first tooltip closes.  I think it would be better to either: a) never collapse a tooltip (and make it take a bit longer to appear), or b) only have them close when you're not hovering over any other tooltip in that section
- sha-2017 should be sha1-2017
- All of the descriptions should be consistent, using "This site" or "This site's".
  - So for SHA1-2016, it would look like: This site's certificate is signed...
  - SHA1-2017 has a typo: "This date is after to the sunset date."
  - null should be: "This site uses TLS but without encryption."  @lgarron: accurate enough for you?
  - mixed: "This page triggers passive mixed content warnings by loading an image from an insecure source."
  - mixed-script: "This page triggers active mixed content warnings by loading a script from an insecure source."
  - upgrade: "This site attempts to load an image over HTTP. If your browser supports the <a href="https://www.w3.org/TR/upgrade-insecure-requests/">upgrade-insecure-requests</a> CSP directive, it will be transparently loaded over HTTPS."
  - subdomain.preloaded-hsts is missing its line break
- We should add the dh4096 and dh8192 sites under Diffie-Hellman

Otherwise, I think it looks great.  @lgarron?

**Comment by Hoverbear** [comment]
Thanks for the keen eyes @marumari :) I'll address these changes in a following commit.

Yes I'm not entirely sure I'm a fan of the tooltip behavior right now. I'm not particularly strong in the CSS animation department. There are delay timers I can work with, I dislike how it shifts down all of the children though. I wonder if I can make the description appear in the button somehow. Hmm...

File: _data/subdomains.yml [hunk 1]
```diff
@@ -168,89 +168,89 @@ dh-small-subgroup:
...
         This site has a very long hostname intended to break browser interfaces.
 pinning-test:
     favicon: red
     background: red
-    category: Misc
+    category: Miscellaneous
     # TODO: Firefox?
     description: |
         This site is preloaded with a bad HPKP pin starting in Chrome 48.
 spoofed-favicon.badssl.com:
```
Comment (jonathanKingston) [inline _data/subdomains.yml]: Should be `spoofed-favicon:` right?

Comment (Hoverbear) [inline _data/subdomains.yml]: Yes!

**Comment by Hoverbear** [comment]
Please let me know if I should rebase this.

**Comment by lgarron** [comment]
@Hoverbear: I think I'm done with my certificate + front page refactoring.  Any chance you have time to rebase this?

I'd like to prioritize landing this now, so I promise I won't make you rebase it anymore after that. :-)

**Comment by lgarron** [comment]
By the way, I'm imagining this for the front page:

screen shot 2016-08-15 at 16 18 44

**Comment by Hoverbear** [comment]
Hey Lucas! I'll try to rebase this soon. :) I have a bit of a backlog because I just moved.

**Comment by Hoverbear** [comment]
Closing this PR since #202 will usurp it. Please don't delete the branch yet since it contains some templating code we may want to use.

**Pull request closed without merging**